pkg/pillar/hypervisor/kvm.go

```diff
@@ -796,6 +807,33 @@ func (ctx KvmContext) CreateDomConfig(domainName string,
 		isVncShimVMEnabled(globalConfig, config)
 	tmplCtx.DomainConfig.DisplayName = domainName
 
+	// if we are not getting the resolution from the cloud-init, check the
+	// global config.
+	fmlResolutions := status.FmlCustomResolution
```

Comment:
Actually, we need the resolution to be known earlier before calling CreateDomConfig. The resolution is used to define the OVMF file at the beginning of the Setup() function...

Reply:
I know, this hasn't been updated with the changes of 9.4 backport.